includes/payments/class-edd-payment.php

```diff
@@ -1606,7 +1606,9 @@ public function update_meta( $meta_key = '', $meta_value = '', $prev_value = ''
 			update_post_meta( $this->ID, '_edd_payment_user_email', $meta_value );
 
 			$current_meta = $this->get_meta();
-			$current_meta['user_info']['email']  = $meta_value;
+			if ( is_array( $current_meta ) ) {
+				$current_meta['user_info']['email']  = $meta_value;
```

Comment:
We need a fallback here I think in the case $current_meta is not an array. Should fall back to the email meta field and then the customer email if the one on payment is empty.